runtime/intern.c

```diff
@@ -224,10 +243,9 @@ static void intern_free_stack(struct caml_intern_state* s)
 {
   if (s->intern_stack != s->intern_stack_init) {
     caml_stat_free(s->intern_stack);
-    /* Reinitialize the globals for next time around */
-    s->intern_stack = s->intern_stack_init;
-    s->intern_stack_limit = s->intern_stack + INTERN_STACK_INIT_SIZE;
   }
+
+  intern_init_stack(s);
```

Comment:
Why is this moved out of the conditional block? You don't need to `intern_init_stack` if the `intern_stack` is not different from `intern_stack_init`.